plugins/modules/custom_image.py

```diff
@@ -0,0 +1,479 @@
+            for r in response_import:
+                if response['uuid'] == r['uuid']:
+                    for key in copy_keys:
+                        response[copy_keys[key]] = r[key]
+            response = [response]
+        else:
+            # More than one image is found
+            for r in response:
+                for i in response_import:
+                    if r['uuid'] == i['uuid']:
+                        for key in copy_keys:
+                            r[copy_keys[key]] = i[key]
+
+        response = self.check_for_uuid(response)
+        return response, info
+
+    def _get(self, api_call):
+        resp, info = self.fetch_both_urls(api_call)
+        if info['status'] == 200:
+            return resp
+        elif info['status'] == 404:
+            return None
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API with GET for '
+                                   '"%s"' % api_call, fetch_url_info=info)
+
+    def _post_or_patch(self, api_call, method, data, filter_none=True):
+        # This helps with tags when we have the full API resource href to update.
+
+        if data is not None:
+            # Sanitize data dictionary
+            # Deepcopy: Duplicate the data object for iteration, because
+            # iterating an object and changing it at the same time is insecure
+            for k, v in deepcopy(data).items():
+                if filter_none and v is None:
+                    del data[k]
+
+            data = self._module.jsonify(data)
+
+        resp, info = self.fetch_both_urls(api_call,
+                                          method=method,
+                                          data=data)
+
+        if info['status'] in (200, 201):
+            return resp
+        elif info['status'] == 204:
+            return None
+        else:
+            self._module.fail_json(msg='Failure while calling the cloudscale.ch API with %s for '
+                                       '"%s".' % (method, api_call), fetch_url_info=info)
+
+    def _post(self, api_call, data=None):
+        if self._module.params['url']:
+            return self._post_or_patch("%s" % api_call, 'POST', data)
+        else:
+            self._module.fail_json(msg="Cannot import a new image without url.")
+
+
+def main():
+    argument_spec = cloudscale_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str'),
+        slug=dict(type='str'),
+        url=dict(type='str'),
+        user_data_handling=dict(type='str',
+                                choices=('pass-through',
+                                         'extend-cloud-config')),
+        uuid=dict(type='str'),
+        tags=dict(type='dict'),
+        state=dict(type='str', default='present',
+                   choices=('present', 'absent')),
+        zones=dict(type='list', elements='str'),
+        source_format=dict(type='str', choices=('raw', )),
```

Comment:
Like we are not restricting the list of possible zones I would also not restrict the list of possible source formats. Without this restriction the module should automatically work with other source formats as they become supported by the API.
```suggestion
        source_format=dict(type='str'),
```